docs/data/data-grid/export/export.md

```diff
@@ -302,6 +302,93 @@ In the following demo, both methods are used to set a custom header and a custom
 
 {{"demo": "ExcelCustomExport.js", "bg": "inline", "defaultCodeOpen": false}}
 
+### Using a web worker
+
+:::warning
+This feature only works with `@mui/styled-engine` v5.11.8 or newer.
+Make sure that the MUI Core version you are using is also installing the correct version for this dependency.
+:::
+
+Instead of generating the Excel file locally, you can delegate the task to a web worker.
+This method reduces the amount of time that the main thread remains frozen, allowing to interact with the grid while the data is exported in background.
+To start using web workers for the Excel export, first you need to create a file with the content below.
+This file will be later used as the worker script, then it's important that it's accessible by a direct URL.
+
+```tsx
+import { setupExcelExportWebWorker } from '@mui/x-data-grid-premium';
+
+setupExcelExportWebWorker();
+```
+
+The final step is to pass the path to the file created to `GridToolbarExport` or the API method:
+
+```tsx
+<GridToolbarExport
+  excelOptions={{
+    worker: () => new Worker('/worker.js'),
```

Comment:
Is there a specific reason why using `js` instead of `ts` here and switching tow codes blocks latter
```suggestion
    worker: () => new Worker('/worker.ts'),
```